psi4/src/psi4/libmints/oeprop.cc

```diff
@@ -1353,44 +1549,60 @@ void OEProp::compute_quadrupole(bool transition)
 
     // Print multipole components
     double dfac = pc_dipmom_au2debye * pc_bohr2angstroms;
-    outfile->Printf( "  %sQuadrupole Moment: [D A]\n", (transition ? "Transition " : ""));
-    outfile->Printf( "    XX: %10.4lf     YY: %10.4lf     ZZ: %10.4lf\n", \
-       qe[0]*dfac, qe[3]*dfac, qe[5]*dfac);
-    outfile->Printf( "    XY: %10.4lf     XZ: %10.4lf     YZ: %10.4lf\n", \
-       qe[1]*dfac, qe[2]*dfac, qe[4]*dfac);
-    outfile->Printf( "\n");
+    if (print_output)
+    {
+        outfile->Printf( "  %sQuadrupole Moment: [D A]\n", (transition ? "Transition " : ""));
+        outfile->Printf( "    XX: %10.4lf     YY: %10.4lf     ZZ: %10.4lf\n", \
+           qe[0]*dfac, qe[3]*dfac, qe[5]*dfac);
+        outfile->Printf( "    XY: %10.4lf     XZ: %10.4lf     YZ: %10.4lf\n", \
+           qe[1]*dfac, qe[2]*dfac, qe[4]*dfac);
+        outfile->Printf( "\n");
+    }
 
     double dtrace = (1.0 / 3.0) * (qe[0] + qe[3] + qe[5]);
-    outfile->Printf( "  Traceless %sQuadrupole Moment: [D A]\n", (transition ? "Transition " : ""));
-    outfile->Printf( "    XX: %10.4lf     YY: %10.4lf     ZZ: %10.4lf\n", \
-       (qe[0]-dtrace)*dfac, (qe[3]-dtrace)*dfac, (qe[5]-dtrace)*dfac);
-    outfile->Printf( "    XY: %10.4lf     XZ: %10.4lf     YZ: %10.4lf\n", \
-       qe[1]*dfac, qe[2]*dfac, qe[4]*dfac);
-    outfile->Printf( "\n");
+    if (print_output)
+    {
+        outfile->Printf( "  Traceless %sQuadrupole Moment: [D A]\n", (transition ? "Transition " : ""));
+        outfile->Printf( "    XX: %10.4lf     YY: %10.4lf     ZZ: %10.4lf\n", \
+           (qe[0]-dtrace)*dfac, (qe[3]-dtrace)*dfac, (qe[5]-dtrace)*dfac);
+        outfile->Printf( "    XY: %10.4lf     XZ: %10.4lf     YZ: %10.4lf\n", \
+           qe[1]*dfac, qe[2]*dfac, qe[4]*dfac);
+        outfile->Printf( "\n");
+    }
 
     // Quadrupole components in Debye Ang
-    std::stringstream s;
-    s << title_ << " QUADRUPOLE XX";
-    Process::environment.globals[s.str()] = qe[0]*dfac;
-    s.str(std::string());
-    s << title_ << " QUADRUPOLE YY";
-    Process::environment.globals[s.str()] = qe[3]*dfac;
-    s.str(std::string());
-    s << title_ << " QUADRUPOLE ZZ";
-    Process::environment.globals[s.str()] = qe[5]*dfac;
-    s.str(std::string());
-    s << title_ << " QUADRUPOLE XY";
-    Process::environment.globals[s.str()] = qe[1]*dfac;
-    s.str(std::string());
-    s << title_ << " QUADRUPOLE XZ";
-    Process::environment.globals[s.str()] = qe[2]*dfac;
-    s.str(std::string());
-    s << title_ << " QUADRUPOLE YZ";
-    Process::environment.globals[s.str()] = qe[4]*dfac;
+    double xx = qe[0]*dfac;
+    double yy = qe[3]*dfac;
+    double zz = qe[5]*dfac;
+    double xy = qe[1]*dfac;
+    double xz = qe[2]*dfac;
+    double yz = qe[4]*dfac;
 
+    auto output = std::make_shared<Matrix>(3,3);
+    Matrix & outmat = *output;
+    outmat.set(0,0,xx);
```

Comment:
In some cases there is no benefit at all. I do it quite often, because it is a null operation and makes my life easier sometimes, when working with operator[].
In case of Matrix and Vector, there are explicit set and get functions, so I can remove it.

On occasion, where it did have some benefit is for example here:
    auto Qa_ptr = std::make_shared<std::vector<double>>(mol->natom());
    auto Qb_ptr = std::make_shared<std::vector<double>>(mol->natom());
    std::vector<double>& Qa = *Qa_ptr;
    std::vector<double>& Qb = *Qb_ptr;
In this case i can just write
Qa[6] instead of (*Qa_ptr)[6].

Should i remove the superfluos occasions?

Reply:
Canonically we would use `Qa_ptr->set(6, value)` or `Qa_ptr->get(6)` and generally rename `Qa_ptr` to `Qa`. In general, we pull out the raw pointer quite a bit. The requested changes are not necessary, just observing current patterns.